File: app/src/main/java/com/kimikevin/eatsplorer/viewmodel/SplashViewModel.java
package com.kimikevin.eatsplorer.viewmodel;

import android.util.Log;
import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;
import java.util.concurrent.ExecutorService;
import java.util.concurrent.Executors;

public class SplashViewModel extends ViewModel {
    private static final String TAG = "SplashViewModel";
    private final MutableLiveData<Boolean> _isDataReady = new MutableLiveData<>(false);

    // Professional Tip: Use ExecutorService for background work instead of raw 'new Thread()'
    private final ExecutorService executorService = Executors.newSingleThreadExecutor();

    public SplashViewModel() {
        checkAppInitialization();
    }

    private void checkAppInitialization() {
        executorService.execute(() -> {
            try {
                // Simulate heavy initialization (e.g., checking user session, warming cache)
                Thread.sleep(1500);
                _isDataReady.postValue(true);
            } catch (InterruptedException e) {
                Log.e(TAG, "Initialization interrupted", e);
                // In case of error, you might still want to proceed or show an error state
                _isDataReady.postValue(true);
            }
        });
    }

    public LiveData<Boolean> isDataReady() {
        return _isDataReady;
    }

    @Override
    protected void onCleared() {
        super.onCleared();
        // Crucial: Clean up threads to prevent memory leaks
        executorService.shutdown();
    }
}

File: app/src/main/java/com/kimikevin/eatsplorer/viewmodel/SplashViewModel.kt
package com.kimikevin.eatsplorer.viewmodel

import androidx.lifecycle.ViewModel
import androidx.lifecycle.viewModelScope
import kotlinx.coroutines.delay
import kotlinx.coroutines.flow.MutableStateFlow
import kotlinx.coroutines.flow.StateFlow
import kotlinx.coroutines.flow.asStateFlow
import kotlinx.coroutines.launch

class SplashViewModel : ViewModel() {
    private val _isDataReady = MutableStateFlow(false)
    val isDataReady: StateFlow<Boolean> = _isDataReady.asStateFlow()

    init {
        checkAppInitialization()
    }

    private fun checkAppInitialization() {
        viewModelScope.launch {
            // Simulate heavy initialization (e.g., checking user session, warming cache)
            delay(1500)
            _isDataReady.value = true
        }
    }
}
